src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -422,11 +423,14 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
    */
   checkForDuplicateInTable (business: any): boolean {
     const checkDuplication = this.getAmalgamatingBusinesses.find((b: any) =>
-      (business.type === AmlTypes.LEAR && b.identifier === business.businessInfo.identifier) ||
-      (business.type === AmlTypes.FOREIGN && b.corpNumber === business.corpNumber)
+      (b.type === AmlTypes.LEAR && b.identifier === business.businessInfo?.identifier) ||
+      (b.type === AmlTypes.FOREIGN && b.corpNumber === business.corpNumber)
     )
 
-    if (checkDuplication) return true
+    if (checkDuplication) {
+      this.isAddingAmalgamatingBusiness = false
```

Comment:
Closing the panel if duplicate was found to stay consistent with the business lookup behavior.